library/std/src/fs.rs

```diff
@@ -271,7 +271,8 @@ pub fn read<P: AsRef<Path>>(path: P) -> io::Result<Vec<u8>> {
 /// Reads the entire contents of a file into a string.
 ///
 /// This is a convenience function for using [`File::open`] and [`read_to_string`]
-/// with fewer imports and without an intermediate variable.
+/// with fewer imports, without an intermediate variable, and with improved performance
+/// by referencing the file's metadata to preallocate buffer size.
```

Comment:
I'm still game to accept this but it's a "should have better performance" or "may be faster and more efficient" rather than "is definitely going to have better performance for (specific implementation detail)" because of OS-specific differences.

Reply:
To clarify, which part are you referring to by saying "may be faster and more efficient"?

Cf. https://github.com/rust-lang/rust/issues/130600#issuecomment-2365289306
> The only optimization that fs::read_to_string has is that it can use the size from the file metadata directly and doesn't need to retrieve the current seek position of the file handle.